scripts/create_xlinear_bertmesh_comparison_csv.py

```diff
@@ -0,0 +1,175 @@
+from xlinear_predict import predict_tags as predict_tags_xlinear
+
+random.seed(42)
+torch.manual_seed(42)
+
+
+def load_mesh_terms_from_file(mesh_terms_list_path):
+    with open(mesh_terms_list_path, "r") as f:
+        mesh_terms = f.readlines()
+    mesh_terms = [term.strip() for term in mesh_terms]
+    return mesh_terms
+
+
+def _extract_data(mesh_elem):
+    # TreeNumberList e.g. A11.118.637.555.567.550.500.100
+    tree_number = mesh_elem[-2][0].text
+    # DescriptorUI e.g. M000616943
+    code = mesh_elem[0].text
+    # DescriptorName e.g. Mucosal-Associated Invariant T Cells
+    name = mesh_elem[1][0].text
+
+    return tree_number, code, name
+
+
+def find_subnames(mesh_metadata_path: str, mesh_terms_list_path: str):
+    """
+    Given a path to a file containing a list of MeSH terms and a path to a file
+    containing MeSH metadata, returns a list of all MeSH terms that are subnames
+    of the MeSH terms in the input file.
+
+    Args:
+        mesh_metadata_path (str): The path to the MeSH metadata file.
+        mesh_terms_list_path (str): The file containing the list of MeSH terms
+                                    to filter by.
+
+    Returns:
+        List[str]: A list of all MeSH terms that are subnames
+                   of the MeSH terms in the input file.
+    """
+    mesh_tree = ET.parse(mesh_metadata_path)
+    mesh_terms = load_mesh_terms_from_file(mesh_terms_list_path)
+    root = mesh_tree.getroot()
+
+    # Do 1st pass to get all their codes
+    top_level_tree_numbers = []
+    pbar = tqdm(root)
+    pbar.set_description("Finding top level tree numbers")
+    for mesh_elem in pbar:
+        try:
+            tree_number, _, name = _extract_data(mesh_elem)
+        except IndexError:
+            continue
+
+        if name not in mesh_terms:
+            continue
+        else:
+            top_level_tree_numbers.append(tree_number)
+
+    # Do 2nd pass to collect all names that are in the same tree as the ones we found
+    all_subnames = []
+    pbar = tqdm(root)
+    pbar.set_description("Finding subnames")
+    for mesh_elem in pbar:
+        try:
+            curr_tree_number, _, name = _extract_data(mesh_elem)
+        except IndexError:
+            continue
+
+        for top_level_tree_number in top_level_tree_numbers:
+            if curr_tree_number.startswith(top_level_tree_number):
+                all_subnames.append(name)
+                break
+
+    return all_subnames
+
+
+def create_comparison_csv(
+    s3_url: str,
+    num_parquet_files_to_consider: int,
+    num_samples_per_cat: int,
+    mesh_metadata_path: str,
+    mesh_terms_list_path: str,
+    bertmesh_path: str,
+    xlinear_path: str,
+    xlinear_label_binarizer_path: str,
+    output_path: str,
+):
+    subnames = find_subnames(mesh_metadata_path, mesh_terms_list_path)
+
+    parquet_files = wr.s3.list_objects(s3_url)
+    random.shuffle(parquet_files)
+
+    all_dfs = []
```

Comment:
let's avoid using `df` or `dict`, `list` etc in names

Reply:
Why? I find it pretty helpful to know what I'm working with